test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,243 @@
+
+            const serviceAddress = await token.interchainTokenService();
+            expect(serviceAddress).to.not.equal(AddressZero);
+
+            const deployerFromSlot = '0x' + updatedSlot0.slice(26);
+            expect(deployerFromSlot.toLowerCase()).to.equal(updatedDeployer.toLowerCase());
+        });
+
+        it('should verify slot 0 is reserved and not overwritten by other variables', async () => {
+            const hyperliquidTokenDeployer = await deployContract(owner, 'InterchainTokenDeployer', [hyperliquidInterchainToken.address]);
+            const salt = ethers.utils.randomBytes(32);
+            const tokenId = ethers.utils.randomBytes(32);
+
+            const newTokenAddress = await hyperliquidTokenDeployer.deployedAddress(salt);
+            const newToken = await ethers.getContractAt('HyperliquidInterchainToken', newTokenAddress, owner);
+
+            await hyperliquidTokenDeployer
+                .deployInterchainToken(salt, tokenId, owner.address, 'TestToken', 'TEST', 18)
+                .then((tx) => tx.wait());
+
+            const newTokenSlot0 = await provider.getStorageAt(newTokenAddress, 0);
+            const newTokenDeployerFromSlot0 = bytes32ToAddress(newTokenSlot0);
+            const deployerFromContract = await newToken.deployer();
+
+            expect(newTokenDeployerFromSlot0).to.equal(AddressZero);
+            expect(deployerFromContract).to.equal(AddressZero);
+
+            const tokenName = await newToken.name();
+            expect(tokenName).to.equal('TestToken');
+
+            const slot0AfterName = await provider.getStorageAt(newTokenAddress, 0);
+            expect(slot0AfterName).to.equal(newTokenSlot0);
+        });
+
+        it('should update deployer with multiple different addresses and verify storage', async () => {
+            const wallets = await ethers.getSigners();
+            const addresses = [owner.address, user.address, AddressZero, wallets[2].address, wallets[3].address];
+
+            for (const addr of addresses) {
+                await token.connect(owner).updateDeployer(addr);
+                expect(await token.deployer()).to.equal(addr);
+
+                const updatedSlot0 = await provider.getStorageAt(token.address, 0);
+                const updatedDeployerFromSlot0 = bytes32ToAddress(updatedSlot0);
+                expect(updatedDeployerFromSlot0.toLowerCase()).to.equal(addr.toLowerCase());
+            }
+
+            await token.connect(owner).updateDeployer(AddressZero);
+            const slot0Zero = await provider.getStorageAt(tokenAddress, 0);
+            const deployerFromSlot0Zero = bytes32ToAddress(slot0Zero);
+            const deployerFromContractZero = await token.deployer();
+
+            expect(deployerFromSlot0Zero.toLowerCase()).to.equal(deployerFromContractZero.toLowerCase());
+            expect(deployerFromContractZero).to.equal(AddressZero);
+        });
+
+        it('should handle authorization and access control', async () => {
```

Comment:
see how we name and do ownership transfer tests here
https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/test/utils/Ownable.js#L10